File: core/src/main/java/com/pluxity/cctv/entity/DeviceCctv.java
package com.pluxity.cctv.entity;

import com.pluxity.device.entity.Device;
import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class DeviceCctv {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "device_id")
    private Device device;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cctv_id")
    private Cctv cctv;

    @Builder
    public DeviceCctv(Device device, Cctv cctv) {
        this.device = device;
        this.cctv = cctv;
    }
}

File: core/src/main/java/com/pluxity/cctv/entity/DeviceCctv.kt
package com.pluxity.cctv.entity

import com.pluxity.device.entity.Device
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
class DeviceCctv(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    @JoinColumn(name = "device_id")
    @ManyToOne(fetch = FetchType.LAZY)
    var device: Device,
    @JoinColumn(name = "cctv_id")
    @ManyToOne(fetch = FetchType.LAZY)
    var cctv: Cctv,
)
